samples/dotnet/github-skills/GitHubSkill.cs

```diff
@@ -0,0 +1,209 @@
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.KernelExtensions;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.Skills.Document;
+using Microsoft.SemanticKernel.Skills.Web;
+
+/// <summary>
+/// Skill for interacting with a GitHub repository.
+/// </summary>
+public class GitHubSkill
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Name of the repository repositoryBranch which will be downloaded and summarized.
+        /// </summary>
+        public const string RepositoryBranch = "repositoryBranch";
+
+        /// <summary>
+        /// The search string to match against the names of files in the repository.
+        /// </summary>
+        public const string SearchPattern = "searchPattern";
+
+        /// <summary>
+        /// Document file path.
+        /// </summary>
+        public const string FilePath = "filePath";
+
+        /// <summary>
+        /// Directory to which to extract compressed file's data.
+        /// </summary>
+        public const string DestinationDirectoryPath = "destinationDirectoryPath";
+
+        /// <summary>
+        /// Name of the memory collection used to store the code summaries.
+        /// </summary>
+        public const string MemoryCollectionName = "memoryCollectionName";
+    }
+
+    /// <summary>
+    /// The max tokens to process in a single semantic function call.
+    /// </summary>
+    private const int MaxTokens = 1024;
+
+    /// <summary>
+    /// The max file size to send directly to memory.
+    /// </summary>
+    private const int MaxFileSize = 2048;
+
+    private readonly ISKFunction _summarizeCodeFunction;
+    private readonly IKernel _kernel;
+    private readonly WebFileDownloadSkill _downloadSkill;
+    private readonly DocumentSkill _documentSkill;
+    private readonly ILogger<GitHubSkill> _logger;
+
+    internal const string SummarizeCodeSnippetDefinition =
+    @"BEGIN CONTENT TO SUMMARIZE:
+{{$INPUT}}
+END CONTENT TO SUMMARIZE.
+
+Summarize the content in 'CONTENT TO SUMMARIZE', identifying main points.
+Do not incorporate other general knowledge.
+Summary is in plain text, in complete sentences, with no markup or tags.
+
+BEGIN SUMMARY:
+";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GitHubSkill"/> class.
+    /// </summary>
+    /// <param name="kernel">Kernel instance</param>
+    /// <param name="downloadSkill">Instance of WebFileDownloadSkill used to download web files</param>
+    /// <param name="documentSkill">Instance of DocumentSkill used to read files</param>
+    /// <param name="logger">Optional logger</param>
+    public GitHubSkill(IKernel kernel, WebFileDownloadSkill downloadSkill, DocumentSkill documentSkill, ILogger<GitHubSkill>? logger = null)
```

Comment:
The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.

Reply:
fixed (otherwise let me know)